articles/Object-Detection-using-Fast-R-CNN-BrainScript.md

```diff
@@ -0,0 +1,537 @@
+Compared to previous work, Fast R-CNN employs a _region of interest pooling_ scheme that allows to reuse the computations from the convolutional layers.
+
+Additional material: a detailed tutorial for object detection using CNTK Fast R-CNN with BrainScript 
+(including optional SVM training and publishing the trained model as a Rest API) can be found [here](https://github.com/Azure/ObjectDetectionUsingCntk).
+
+## Setup
+
+To run the code in this example, you need a CNTK Python environment (see [here](./Setup-CNTK-on-your-machine.md) for setup help). You need to work from your Python 3.4 environment (if you are using Anaconda Python type `activate cntk-py34` (from a standard command line, not PowerShell), assuming `cntk-py34` is your environment name). Further you need to install a few additional packages. From your Python 3.4 environment (64bit version assumed, Python 3.5 analogously), go to the FastRCNN folder and run:
+
+```
+pip install -r requirements.txt
+```
+
+Known issue: to install scikit-learn you might have to run `conda install scikit-learn` if you use Anaconda Python.
+You will further need Scikit-Image and OpenCV to run these examples (and possibly numpy and scipy if your Python 3.4 package does not come with them).
+You need to download the corresponding wheel packages and install them manually. On Linux you can `conda install scikit-image opencv`.
+For Windows users, visit http://www.lfd.uci.edu/~gohlke/pythonlibs/, and download:
+
+* Python 3.4
+  * scikit_image-0.12.3-cp34-cp34m-win_amd64.whl
+  * opencv_python-3.1.0-cp34-cp34m-win_amd64.whl
+* Python 3.5
+  * scikit_image-0.12.3-cp35-cp35m-win_amd64.whl
+  * opencv_python-3.2.0-cp35-cp35m-win_amd64.whl
+
+Once you download the respective wheel binaries, install them with (Python 3.5 analogously):
+
+```
+pip install your_download_folder/scikit_image-0.12.3-cp34-cp34m-win_amd64.whl
+```
+
+Remark 1: if you see the message _No module named past_ when running the scripts please execute `pip install future`.
+<br>Remark 2: If you have a Python 3.5 environment you need the corresponding _cp35_ wheels.
+
+This tutorial code assumes you are using 64bit version of Python, either 3.4 or 3.5 on Windows or 3.4 on Linux, since the required Fast R-CNN DLL files under [utils](https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Detection/FastRCNN/BrainScript/fastRCNN/utils)
+are prebuilt for those versions. If your task requires the use of a different Python version, please recompile these DLL files yourself in the correct environment (see [below](#pre-compiled-binaries-for-bounding-box-regression-and-non-maximum-suppression)).
+
+The tutorial further assumes that **the folder where cntk.exe resides is in your PATH environment variable**. (To add the folder to your PATH you can run the following command from a command line (assuming the folder where cntk.exe is on your machine is C:\src\CNTK\x64\Release): `set PATH=C:\src\CNTK\x64\Release;%PATH%`.)
```

Comment:
>The tutorial further assumes that **the folder where cn [](start = 0, length = 55)

isn't that whole thing part of the python setup as well as the activation?

Reply:
We had numerous people asking about issues related to 'cannot find cntk.exe' or the like. That's why we added this sentence in bold to the setup section (months ago btw).